neps/nep-0452.md

```diff
@@ -0,0 +1,239 @@
+///   The key types currently supported are secp256k1 and ed25519. Ed25519 public keys accepted are 32 bytes
+///   and secp256k1 keys are the uncompressed 64 format.
+///
+/// Arguments:
+/// * `new_account_id`: the valid NEAR account which is being created and should
+///   receive the linkdrop assets
+/// * `new_public_key`: the valid public key that should be used for the access key added to the newly created account (serialized with borsh).
+///
+/// Returns `true` if the claim was successful meaning the `new_account_id` was created and all assets were sent to it.
+function create_account_and_claim(new_account_id: string, new_public_key: string) -> Promise<boolean>;
+```
+
+## Reference Implementation
+
+Below are some references for linkdrop contracts.
+
+- [Link Drop Contract](https://github.com/near/near-linkdrop)
+
+- [Keypom Contract](https://github.com/keypom/keypom)
+
+## Security Implications
+
+1. Linkdrop Creation
+   Linkdrop creation involves creating keypairs that, when used, have access to assets such as $NEAR, FTs, NFTs, etc. These keys should be limited access and restricted to specific functionality. For example, they should only have permission to call `claim` and `create_account_and_claim`. Since the keys allow the holder to sign transactions on behalf of the linkdrop contract, without the proper security measures, they could be used in a malicious manner (for example executing private methods or owner-only functions).
+
+Another important security implication of linkdrop creation is to ensure that only one key is mapped to a set of assets at any given time. Externally, assets such as FTs, and NFTs belong to the overall linkdrop contract account rather than a specific access key. It is important to ensure that specific keys can only claim assets that they are mapped to.
+
+2. Linkdrop Key Management  
+   Key management is a critical safety component of linkdrops. The linkdrop contract should implement a key management strategy for keys such that a reentrancy attack does not occur. For example, one strategy may be to "lock" or mark a key as "in transfer" such that it cannot be used again until the transfer is complete.
+
+3. Asset Refunds & Failed Claims
+   Given that linkdrops could contain multiple different assets such as NFTs, or fungible tokens, sending assets might happen across multiple blocks. If the claim was unsuccessful (such as passing in an invalid account ID), it is important to ensure that all state is properly managed and assets are optionally refunded depending on the linkdrop contract's implementation.
+
+4. Fungible Tokens & Future Data
+   Fungible token contracts require that anyone receiving tokens must be registered. For this reason, it is important to ensure that storage for accounts claiming linkdrops is paid for. This concept can be extended to any future data types that may be added. You must ensure that all the pre-requisite conditions have been met for the asset that is being transferred.
```

Comment:
> ensure that storage for accounts claiming linkdrops is paid for

That may not be possible, because if it's a new account claiming a drop, then we don't know the new account address.